pkg/azureutils/informer_common.go

```diff
@@ -0,0 +1,145 @@
+limitations under the License.
+*/
+
+package azureutils
+
+import (
+	"context"
+
+	crdInformers "k8s.io/apiextensions-apiserver/pkg/client/informers/externalversions"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/klog/v2"
+	azdiskinformers "sigs.k8s.io/azuredisk-csi-driver/pkg/apis/client/informers/externalversions"
+)
+
+type InformerFactory interface {
+	Start(stopCh <-chan struct{})
+	Informer() cache.SharedIndexInformer
+}
+
+type azNodeInformer struct {
+	factory  azdiskinformers.SharedInformerFactory
+	informer cache.SharedIndexInformer
+}
+
+func NewAzNodeInformer(factory azdiskinformers.SharedInformerFactory) InformerFactory {
+	return &azNodeInformer{
+		factory:  factory,
+		informer: factory.Disk().V1beta2().AzDriverNodes().Informer(),
+	}
+}
+
+func (i *azNodeInformer) Start(stopCh <-chan struct{}) {
+	i.factory.Start(stopCh)
+}
+
+func (i *azNodeInformer) Informer() cache.SharedIndexInformer {
+	return i.informer
+}
+
+type azVolumeAttachmentInformer struct {
+	factory  azdiskinformers.SharedInformerFactory
+	informer cache.SharedIndexInformer
+}
+
+func NewAzVolumeAttachmentInformer(factory azdiskinformers.SharedInformerFactory) InformerFactory {
+	return &azVolumeAttachmentInformer{
+		factory:  factory,
+		informer: factory.Disk().V1beta2().AzVolumeAttachments().Informer(),
+	}
+}
+
+func (i *azVolumeAttachmentInformer) Start(stopCh <-chan struct{}) {
+	i.factory.Start(stopCh)
+}
+
+func (i *azVolumeAttachmentInformer) Informer() cache.SharedIndexInformer {
+	return i.informer
+}
+
+type azVolumeInformer struct {
+	factory  azdiskinformers.SharedInformerFactory
+	informer cache.SharedIndexInformer
+}
+
+func NewAzVolumeInformer(factory azdiskinformers.SharedInformerFactory) InformerFactory {
+	return &azVolumeInformer{
+		factory:  factory,
+		informer: factory.Disk().V1beta2().AzVolumes().Informer(),
+	}
+}
+
+func (i *azVolumeInformer) Start(stopCh <-chan struct{}) {
+	i.factory.Start(stopCh)
+}
+
+func (i *azVolumeInformer) Informer() cache.SharedIndexInformer {
+	return i.informer
+}
+
+type nodeInformer struct {
+	factory  informers.SharedInformerFactory
+	informer cache.SharedIndexInformer
+}
+
+func NewNodeInformer(factory informers.SharedInformerFactory) InformerFactory {
+	return &nodeInformer{
+		factory:  factory,
+		informer: factory.Core().V1().Nodes().Informer(),
+	}
+}
+
+func (i *nodeInformer) Start(stopCh <-chan struct{}) {
+	i.factory.Start(stopCh)
+}
+
+func (i *nodeInformer) Informer() cache.SharedIndexInformer {
+	return i.informer
+}
+
+type crdInformer struct {
+	factory  crdInformers.SharedInformerFactory
+	informer cache.SharedIndexInformer
+}
+
+func NewCrdInformer(factory crdInformers.SharedInformerFactory) InformerFactory {
+	return &crdInformer{
+		factory:  factory,
+		informer: factory.Apiextensions().V1().CustomResourceDefinitions().Informer(),
+	}
+}
+
+func (i *crdInformer) Start(stopCh <-chan struct{}) {
+	i.factory.Start(stopCh)
```

Comment:
I think we should either start it in a goroutine here, or return `i.factory.Start(stopCh)` and start in a goroutine where it is called.

Reply:
`Start` method already starts each informer in a go routine [here](https://github.com/kubernetes-sigs/azuredisk-csi-driver/blob/91e9fed829ea6447979156004bd4174b7f51f0d5/pkg/apis/client/informers/externalversions/factory.go#L117), do we still run this method in another go routine?

Reply:
There's no need to call `Start` in a goroutine since it creates one in its implementation.

Reply:
Thanks for confirming.